packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -135,29 +166,74 @@ export interface TreeView<TSchema extends ImplicitFieldSchema> extends IDisposab
 	 */
 	upgradeSchema(): void;
 
+	/**
+	 * Initialize the tree, setting the stored schema to match this view's schema and setting the tree content.
+	 *
+	 * Only valid to call when this view's {@link SchemaCompatibilityStatus.canInitialize} is true.
+	 *
+	 * Applications should typically call this function before attaching a `SharedTree`.
+	 * @param content - The content to initialize the tree with.
+	 */
+	initialize(content: InsertableTreeFieldFromImplicitField<TSchema>): void;
+
 	/**
 	 * Events for the tree.
 	 */
 	readonly events: ISubscribable<TreeViewEvents>;
 }
 
 /**
- * Information about how a view schema was incompatible.
+ * Information about a view schema's compatibility with the document's stored schema.
  * @public
  */
-export interface SchemaIncompatible {
+export interface SchemaCompatibilityStatus {
 	/**
-	 * True iff the view schema supports all possible documents permitted by the stored schema.
+	 * Whether the view schema is an exact match to the stored schema.
+	 */
+	readonly isExactMatch: boolean;
+
+	/**
+	 * Whether the current view schema is sufficiently compatible with the stored schema to allow viewing tree data.
+	 * This is true when the documents allowed by the view schema are a subset of those allowed by the stored schema.
```

Comment:
This sounds like its using "subset" meaning that a portion of a document's data is describable by the view schema. In all my logic I always used subset to relate to the sets of possible in schema documents allowed by different schema ( ex: `number | string` allows a superset of whats allowed by `number`) while I think your use here is the is something different which I'm unclear on the details of.

Reply:
Hmm, that usage matches my intent here. Anything I can do to make it clearer?

Reply:
If viewing some data might throw (as documented below), I don't think we should say "canView" is true, especially if what might throw isn't documented as throwing recoverable exceptions. Maybe we should have "mayAttemptViewing" for cases where reading might throw?

My interpretation of "Whether the current view schema is sufficiently compatible with the stored schema to allow viewing tree data." is that if true, then you can view the tree (implicitly all of the tree) with no exceptions thrown.

Maybe we can have "mayAttemptViewing", "canViewSubset", "canViewAll" and "writable"  as separate things?

Also I think its worth noting that in our current setup, a document can be writable, and readable (fully compatible) without matching exactly. I don't know if we actually detect it correctly, but cases like having extra unused schema or use of forbidden fields/never nodes, and in the future having the right adapters can lead to non-exact match cases being fully compatible.

For example a schema with an optional field thats impossible to put and a different schema with the same optional field thats impossible to put anything in can be compatible for read and write, even if the effective never field in one is expressed by having no valid types, and the other by having a valid type which can not exist (ex: has an impossible field, like a value with with no allowed types).

This is why I didn't call out exactMatching as a thing in the API, only compatibility. View schema and stored schema don't even have the same type, so exactMatch wouldn't be super well defined anyway.

Reply:
> If viewing some data might throw (as documented below), I don't think we should say "canView" is true, especially if what might throw isn't documented as throwing recoverable exceptions. Maybe we should have "mayAttemptViewing" for cases where reading might throw?

Fair feedback--I believe I've taken this into account w.r.t new direction of the PR.

> My interpretation of "Whether the current view schema is sufficiently compatible with the stored schema to allow viewing tree data." is that if true, then you can view the tree (implicitly all of the tree) with no exceptions thrown.

Agreed

> Maybe we can have "mayAttemptViewing", "canViewSubset", "canViewAll" and "writable" as separate things?

I'd prefer to keep the API surface proposed here and improve its semantics instead of adding more complicated concepts. See updated doc comment on canView.

> Also I think its worth noting that in our current setup, a document can be writable, and readable (fully compatible) without matching exactly. I don't know if we actually detect it correctly, but cases like having extra unused schema or use of forbidden fields/never nodes, and in the future having the right adapters can lead to non-exact match cases being fully compatible.

Yeah, I added some doc to `SchemaCompatibilityStatus` noting these cases and explicitly noting matches are in terms of allowed document sets. I think practically speaking these cases don't matter much to applications.